File: 2063_vowels-of-all-substrings/csharp/solution.cs
public class Solution
{
    public long CountVowels(string word)
    {
        var n = word.Length;
        long total = 0;

        bool exists(char character)
        {
            switch (character)
            {
                case 'a':
                case 'e':
                case 'i':
                case 'o':
                case 'u':
                    return true;
                default:
                    return false;
            }
        }

        for (int i = 0; i < n; i++)
        {
            if (exists(word[i]))
            {
                total += (long)(i + 1) * (n - i);
            }
        }

        return total;
    }
}

File: 2063_vowels-of-all-substrings/rust/solution.rs
#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn count_vowels(word: String) -> i64 {
        let n = word.len();
        let mut total: i64 = 0;

        fn exists(character: char) -> bool {
            match character {
                'a' | 'e' | 'i' | 'o' | 'u' => true,
                _ => false,
            }
        }

        for i in 0..n {
            if exists(word.as_bytes()[i] as char) {
                total += ((i + 1) as i64) * ((n - i) as i64);
            }
        }

        total
    }
}
